### PR TITLE
weather_status: Improve contrast of favorite icon

### LINKED ISSUE #36969: [BITV] 9.1.4.11/2.3 - "Weather" menu - The "Favorites" button does not meet the contrast requirements for graphical controls, whether it is active or inactive. (1)
- [ ] Fix
- [ ] Backport

20ff47cad6376bf69aae1fdaf0a50b3e

<details>
<summary>Details</summary>

https://report.bitvtest.de/default-en/d63601ac-cb34-4645-8256-66bec78964a0.html#checkpoint-3dbfb7615a-v2-n3

</details>

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: #36969

## Summary
Do not use a yellow icon but the default colored one and use the `StarOff` icon for removing a favorite instead of using `StarOutline`.

This also adds a feature: The current selected favorite is highlighted by setting the icon color to the primary color.

## screenshots
before | after
---|---
image|image
image|image
image|image

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Screenshots before/after for front-end changes

### CODE REVIEW

**Comment by szaimen** [comment]
@nimishavijay @jancborchardt I think we need to think here how to properly display the current state: enabled <-> disabled

**Comment by susnux** [comment]
> @nimishavijay @jancborchardt I think we need to think here how to properly display the current state: enabled <-> disabled

Yes, mainly we should  decide whether we want to keep the yellow star or not, because we have the same problem on the files list (yellow star on white background => too low contrast).

**Review by jancborchardt — CHANGES_REQUESTED** [review]
StarOff looks strange, cause it looks as if a click on it would remove the star.

Instead we should use the "non-filled" star icon for the state when it's not a favorite, that is more common.

Regarding color, just dark is fine, or do we find maybe a dark yellow/gold that still looks like a star color? In dark mode it can/must of course be light yellow.

**Comment by susnux** [comment]
> do we find maybe a dark yellow/gold that still looks like a star color

@jancborchardt yellow will not work on white background, but dark golden at least fulfills the AA criteria. Will look like this:
image
image

Do you think this looks better than the default color?

**Comment by jancborchardt** [comment]
@susnux yep, I would say that dark gold is good! It at least conveys a little bit of a star color, otherwise favorite gets lost in the sea of other functions.

(Different issue, but we should probably also make sure that the "Delete" icon in all the action menus is colored accordingly – also something for the Penpot library @nimishavijay )

**Review by jancborchardt — APPROVED** [review]

File: apps/weather_status/src/App.vue [hunk 1]
```diff
@@ -602,6 +609,17 @@ export default {
     min-height: 44px !important;
 }
 
+// Set color to primary element for current / active favorite address
+.current-favorite {
+	//color: var(--color-primary-element)!important;
```
Comment (Pytal) [inline apps/weather_status/src/App.vue:614]:
```suggestion
```

**Review by Pytal — APPROVED** [review]

**Comment by susnux** [comment]
I had to adjust the color slightly to `#a08b00` to guarantee the contrast even when hovered.

**Review by nfebe — APPROVED** [review]

**Comment by szaimen** [comment]
Conflicts

**Comment by susnux** [comment]
> Conflicts

Resolved

**Pull request merged**